src/hypercore-file-logger.js

```diff
@@ -50,43 +52,58 @@ class HyperCoreFileLogger extends HyperCoreLogger {
    * @param {boolean} [swarmOpts.multiplex]
    */
   constructor (
-    filepath, republish,
+    pathlike, republish,
     feedDir, feedKey = null, feedOpts = null, swarmOpts = null
   ) {
     super(feedDir, feedKey, feedOpts, swarmOpts)
 
     this.republish = republish
-    this.filepath = filepath
+    this.pathlike = pathlike
 
-    this.tail = new Tail(this.filepath)
-    this.tail.on('line', (line) => {
-      const data = line + '\n'
-      this.feed.append(data)
-    })
+    /** @type {Object<string, Tail>} */
+    this.fileTails = {}
   }
 
   async start () {
+    const files = await resolvePaths(this.pathlike)
+    if (!files.length) throw new Error('ERR_FILE_NOT_FOUND')
+
     await super.start()
-    if (this.republish) {
-      const encoding = this.feedOpts.valueEncoding
-      const rstream = fs.createReadStream(this.filepath, { encoding })
 
-      const rl = readline.createInterface({
-        input: rstream,
-        crlfDelay: Infinity
-      })
+    await Promise.all(files.map(async (file) => {
+      const prefix = files.length > 1 ? file + ' >>> ' : ''
+
+      if (this.republish) {
+        const encoding = this.feedOpts.valueEncoding
+        const rstream = fs.createReadStream(file, { encoding })
+
+        const rl = readline.createInterface({
+          input: rstream,
+          crlfDelay: Infinity
+        })
 
-      for await (const line of rl) {
-        this.feed.append(line + '\n')
+        for await (const line of rl) {
+          this.feed.append(prefix + line + '\n')
```

Comment:
probably you want to use `promisify` on append or use the callback to avoid any race conditions